test/unit/product/product-component.js

```diff
@@ -3,715 +3,926 @@ import Checkout from '../../../src/components/checkout';
+              options: configCopy.options,
+            }, {
+              client: ShopifyBuy.buildClient({
+                domain: 'test.myshopify.com',
+                storefrontAccessToken: 123,
+              }),
+            });
+            productFetchByHandleStub = sinon.stub(handleProduct.props.client.product, 'fetchByHandle').resolves({});
+          });
+
+          afterEach(() => {
+            productFetchByHandleStub.restore();
+          });
+
+          it('calls fetchProductByHandle with product handle', async () => {
+            await handleProduct.sdkFetch();
+            assert.calledWith(productFetchByHandleStub, 'hat');
+          });
+        });
+
+        it('rejects if there is no storefrontId or handle', async () => {
+          const errProduct = new Product({
             options: configCopy.options,
           }, {
             client: ShopifyBuy.buildClient({
               domain: 'test.myshopify.com',
               storefrontAccessToken: 123,
             }),
           });
-          productFetchByHandleStub = sinon.stub(handleProduct.props.client.product, 'fetchByHandle').resolves({});
-        });
+          errProduct.storefrontId = null;
+          errProduct.handle = null;
 
-        afterEach(() => {
-          productFetchByHandleStub.restore();
-        });
-
-        it('calls fetchProductByHandle with product handle', () => {
-          handleProduct.sdkFetch();
-          assert.calledWith(productFetchByHandleStub, 'hat');
+          try {
+            await errProduct.sdkFetch();
+            assert.fail();
+          } catch (err) {
+            assert.equal(err.message, 'SDK Fetch Failed');
+          }
         });
       });
     });
 
-    describe('onButtonClick()', () => {
-      beforeEach(async () => {
-        const newProduct = await product.init(testProductCopy);
-        newProduct.cart.model.lineItems = [];
-        newProduct.cart.props.client = newProduct.props.client;
+    describe('non fetch methods', () => {
+      beforeEach(() => {
+        sinon.stub(props.client.shop, 'fetchInfo').resolves(shopFixture);
+        sinon.stub(props.client.product, 'fetch').resolves(productFixture);
+        product = new Product(configCopy, props);
       });
 
-      it('add variant to cart is called with the right quantity of selected variant', () => {
-        product.selectedQuantity = 1111;
-        const addToCartStub = sinon.stub(product.cart, 'addVariantToCart');
-        const evt = new Event('click shopify-buy__btn--parent');
-        const target = 'shopify-buy__btn--parent';
+      describe('init()', () => {
+        let createCartStub;
+        let renderStub;
 
-        product.onButtonClick(evt, target);
-        assert.calledOnce(addToCartStub);
-        assert.calledWith(addToCartStub, product.selectedVariant, 1111);
-        addToCartStub.restore();
-      });
+        beforeEach(() => {
+          createCartStub = sinon.stub(product.props, 'createCart').resolves('test');
```

Comment:
Can we change the value of the resolved value to something that resembles a cart object? The use of `test` is a bit confusing since `init` is also called with `test` as the `data` param.